stetho/src/main/java/com/facebook/stetho/inspector/protocol/ScreencastDispatcher.java

```diff
@@ -0,0 +1,131 @@
+ */
+
+package com.facebook.stetho.inspector.protocol;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Looper;
+import android.util.Base64;
+import android.view.View;
+import com.facebook.stetho.common.LogUtil;
+import com.facebook.stetho.inspector.elements.android.ActivityTracker;
+import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
+import com.facebook.stetho.json.annotation.JsonProperty;
+
+import java.io.ByteArrayOutputStream;
+
+public class ScreencastDispatcher {
+  private final Handler mMainHandler = new Handler(Looper.getMainLooper());
+  private final HandlerThread mHandlerThread = new HandlerThread("Screencast Thread");
+  private final BitmapFetchRunnable mBitmapFetchRunnable = new BitmapFetchRunnable();
+  private final ActivityTracker mActivityTracker = ActivityTracker.get();
+  private final EventDispatchRunnable mEventDispatchRunnable = new EventDispatchRunnable();
+
+  private Handler mBackgroundHandler;
+  private JsonRpcPeer mPeer;
+
+  public ScreencastDispatcher() {
+    mHandlerThread.start();
+  }
+
+  public void startScreencast(JsonRpcPeer peer) {
+    LogUtil.d("Starting screencast");
+    mPeer = peer;
+    // Send about 30 frames per second
+    mBackgroundHandler = new Handler(mHandlerThread.getLooper());
+    mMainHandler.postDelayed(mBitmapFetchRunnable, 33);
+  }
+
+  public void stopScreencast() {
+    LogUtil.d("Stopping screencast");
+    mMainHandler.removeCallbacks(mBitmapFetchRunnable);
+    mBackgroundHandler.removeCallbacks(mEventDispatchRunnable);
+  }
+
+  private class BitmapFetchRunnable implements Runnable {
+    @Override
+    public void run() {
+      mBackgroundHandler.post(mEventDispatchRunnable
+          .withBitmap(getScreenBitmap())
+          .withEndAction(this));
+    }
+
+    private Bitmap getScreenBitmap() {
+      Activity activity = mActivityTracker.tryGetTopActivity();
+      if (activity == null) {
+        return null;
+      }
+      View rootView = activity.getWindow().getDecorView();
+      rootView.setDrawingCacheEnabled(true);
+      Bitmap drawingCache = rootView.getDrawingCache();
+      Bitmap bitmap = android.graphics.Bitmap.createBitmap(drawingCache);
+      rootView.setDrawingCacheEnabled(false);
+      return bitmap;
+    }
+  }
+
+  private class EventDispatchRunnable implements Runnable {
+    private Bitmap mBitmap;
+    private Runnable mEndAction;
+
+    private EventDispatchRunnable withBitmap(Bitmap bitmap) {
+      mBitmap = bitmap;
+      return this;
+    }
+
+    private EventDispatchRunnable withEndAction(Runnable endAction) {
+      mEndAction = endAction;
+      return this;
+    }
+
+    @Override
+    public void run() {
+      ScreencastFrameEvent screencastFrameEvent = new ScreencastFrameEvent();
+      ByteArrayOutputStream stream = new ByteArrayOutputStream();
+      mBitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
```

Comment:
Chrome DevTools prefixes the base64 encoded url with [``data:image/jpg;base64``](https://github.com/auchenberg/chrome-devtools-app/blob/master/app/devtools/front_end/screencast/ScreencastView.js#L152), so you should probably compress the image as JPEG.